src/Collector.cs

```diff
@@ -79,25 +83,34 @@ private void StartBackgroundWorker()
 
     private void WorkerOnDoWork(object sender, DoWorkEventArgs doWorkEventArgs)
     {
+      var failures = 0;
+      Socket socket = null;
+
+      // This is not really the worker loop, it is just very careful connect/authenticate
+      // If you want the worker loop, look at SendQueuedMessages
       while (!_worker.CancellationPending)
         {
-          Socket socket = null;
-          var failures = 0;
-
           try
             {
               socket = Connect();
-              Authenticate(socket);
-              failures = 0;
+              if(!Authenticate(socket))
+                {
+                  socket.Close();
+                  Thread.Sleep(5000);
```

Comment:
Yes.  Without this sleep, doing testing with ncat, I my entire computer filled with "hello version ___" messages in approximately 0.1 seconds.  Some sort of authenticate throttle is definitely needed.  Ultimately, I was tempted to just throw here, since the rest of this class happily uses exceptions for flow control, but I couldn't quite get there.